oidc_middleware.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+// Largely taken from https://developer.okta.com/blog/2021/01/04/offline-jwt-validation-with-go
+
+func GetOidcPublicKeys(config OICDConfig) *map[string]*rsa.PublicKey {
```

Comment:
Doesn't seem like this needs to be exported.
Also, `map` is passed the same way as slices (basically by reference), so you don't need to return/pass a pointer to it.

```suggestion
func getOidcPublicKeys(config OICDConfig) map[string]*rsa.PublicKey {
```

Reply:
Weird that this ended up here. Was meant to be in the review below...

Reply:
Continuing discussion on rsaKeys pointer or not below.